staging/src/k8s.io/apimachinery/pkg/util/wait/loop_test.go

```diff
@@ -0,0 +1,445 @@
+			callback: func(attempts int) (bool, error) {
+				return attempts >= 100, nil
+			},
+			attemptsExpected: 100,
+		},
+		{
+			name: "condition returns error even if ok is true",
+			callback: func(_ int) (bool, error) {
+				return true, conditionErr
+			},
+			attemptsExpected: 1,
+			errExpected:      conditionErr,
+		},
+		{
+			name: "condition exits after an error",
+			callback: func(_ int) (bool, error) {
+				return false, conditionErr
+			},
+			attemptsExpected: 1,
+			errExpected:      conditionErr,
+		},
+		{
+			name:             "context already canceled no attempts expected",
+			context:          cancelledContext,
+			callback:         defaultCallback,
+			attemptsExpected: 0,
+			errExpected:      context.Canceled,
+		},
+		{
+			name:               "context cancelled after 5 attempts",
+			context:            defaultContext,
+			callback:           defaultCallback,
+			cancelContextAfter: 5,
+			attemptsExpected:   5,
+			errExpected:        context.Canceled,
+		},
+		{
+			name:             "context at deadline no attempts expected",
+			context:          deadlinedContext,
+			callback:         defaultCallback,
+			attemptsExpected: 0,
+			errExpected:      context.DeadlineExceeded,
+		},
+	}
+
+	for _, test := range tests {
+		for _, immediate := range []bool{true, false} {
+			t.Run(fmt.Sprintf("immediate=%t", immediate), func(t *testing.T) {
+				for _, sliding := range []bool{true, false} {
+					t.Run(fmt.Sprintf("sliding=%t", sliding), func(t *testing.T) {
+						t.Run(test.name, func(t *testing.T) {
+							contextFn := test.context
+							if contextFn == nil {
+								contextFn = defaultContext
+							}
+							ctx, cancel := contextFn()
+							defer cancel()
+
+							timer := Backoff{Duration: time.Microsecond}.Timer()
+							attempts := 0
+							err := loopConditionUntilContext(ctx, timer, test.immediate, test.sliding, func(_ context.Context) (bool, error) {
+								attempts++
+								defer func() {
+									if test.cancelContextAfter > 0 && test.cancelContextAfter == attempts {
+										cancel()
+									}
+								}()
+								return test.callback(attempts)
+							})
+
+							if test.errExpected != err {
+								t.Errorf("expected error: %v but got: %v", test.errExpected, err)
+							}
+
+							if test.attemptsExpected != attempts {
+								t.Errorf("expected attempts count: %d but got: %d", test.attemptsExpected, attempts)
+							}
+						})
+					})
+				}
+			})
+		}
+	}
+}
+
+type timerWrapper struct {
+	timer   clock.Timer
+	resets  []time.Duration
+	onReset func(d time.Duration)
+}
+
+func (w *timerWrapper) C() <-chan time.Time { return w.timer.C() }
+func (w *timerWrapper) Stop() bool          { return w.timer.Stop() }
+func (w *timerWrapper) Reset(d time.Duration) bool {
+	w.resets = append(w.resets, d)
+	b := w.timer.Reset(d)
+	if w.onReset != nil {
+		w.onReset(d)
+	}
+	return b
+}
+
+func Test_loopConditionUntilContext_timings(t *testing.T) {
```

Comment:
I'm not able to understand this test :(

It seems you've been using it during development to test something , because have some arguments that are not used like `context`or `errExpected`or always with the same value like  `callback`